file.go

```diff
@@ -79,6 +81,51 @@ func (j *JSONLoader) Load(s interface{}) error {
 	return json.NewDecoder(r).Decode(s)
 }
 
+// YAMLLoader satisifies the loader interface. It loads the configuration from
+// the given yaml file
+type YAMLLoader struct {
+	Path   string
+	Reader io.Reader
+}
+
+// Load loads the source into the config defined by struct s
+// Defaults to using the Reader if provided, otherwise tries to read from the
+// file
+func (y *YAMLLoader) Load(s interface{}) error {
+	var data []byte
+	var err error
+
+	if y.Reader != nil {
+		data, err = ioutil.ReadAll(y.Reader)
+
+	} else if y.Path != "" {
+
+		var file *os.File
+
+		file, err = getConfig(y.Path)
+		if err != nil {
+			return err
+		}
+
+		data, err = ioutil.ReadAll(file)
+
+		file.Close()
```

Comment:
@rjeczalik what i remember is, it wasn't working properly with relative paths, so we added this. but i dont remember in which case it wasnt working. i've checked go release history but couldn't find anything regarding any fix might relate to this